### PR TITLE
Refactor queries

### PR DESCRIPTION
**Pull request by mayabrandi**
This PR is the result of me and Måns finding ways to make more efficient queries to the database wich speeds up the views and makes the code more clean. 

The Important changes are in:

**vogue/adapter/plugin.py** Adding aggregation function
**vogue/server/utils.py** Removing lots of functions and adding four new:
* pipe_value_per_month() - Builds the pipe to pass to the aggregation function
* reformat_aggregate_results() - Function to reformat the aggregated results
* value_per_month() - wrapper function for the above functions.
* plot_atributes() - Small function to format some stuff for the plots. (title etc)

**vogue/server/views.py** The plot views are now calling value_per_month() instead of find_key_over_time(), which is deleted. The output from value_per_month is a little different wich is why some things have changes in the view funcitons.

Some more changes in:

**<..>.html** all htmls conaining plots are updated based on the new data format
**vogue/constants/constants.py** Adding test samples that need to be filtered out from all plots

### CODE REVIEW

File: vogue/adapter/plugin.py [hunk 1]
```diff
@@ -77,3 +77,6 @@ def analysis(self, analysis_id: str):
     def find_samples(self, query:dict)-> list:
         samples = self.sample_collection.find(query)
         return list(samples)
+        
+    def samples_aggregate(self, pipe : list):
```
Comment (moonso) [inline vogue/adapter/plugin.py]: Could you add docstrings to these?

File: vogue/server/templates/lucigen.html [hunk 2]
```diff
@@ -62,16 +62,16 @@
...
-    {% if amount_concentration_time.group.no_group is defined %}
-    var data = {{ amount_concentration_time.group.no_group.data | tojson }}
+    {% if amount_concentration_time.all_samples is defined %}
+    var data = {{ amount_concentration_time.all_samples.data | tojson }}
     {% endif %}
 
     Highcharts.chart('amount_concentration_time', {
         chart: { type: 'spline' },
-        title: { text: '{{amount_concentration_time.title}}'},
+        title: { text: '{{plot_atributes.title}}'},
```
Comment (moonso) [inline vogue/server/templates/lucigen.html]:
```suggestion
        title: { text: '{{plot_attributes.title}}'},
```

File: vogue/server/templates/lucigen.html [hunk 3]
```diff
@@ -62,16 +62,16 @@
...
         subtitle: { text: '' },
-        xAxis: { categories: {{ amount_concentration_time.labels| tojson }}},
-        yAxis: { title: { text: {{ amount_concentration_time.axis.y | tojson }}}},
+        xAxis: { categories: {{ plot_atributes.labels| tojson }}},
```
Comment (moonso) [inline vogue/server/templates/lucigen.html]:
```suggestion
        xAxis: { categories: {{ plot_attributes.labels| tojson }}},
```

File: vogue/server/templates/lucigen.html [hunk 4]
```diff
@@ -62,16 +62,16 @@
...
+        yAxis: { title: { text: {{ plot_atributes.axis.y | tojson }}}},
```
Comment (moonso) [inline vogue/server/templates/lucigen.html]:
```suggestion
        yAxis: { title: { text: {{ plot_attributes.axis.y | tojson }}}},
```

File: vogue/server/templates/microbial.html [hunk 5]
```diff
@@ -30,11 +30,11 @@
             type: 'spline'
         },
         title: {
-            text: '{{microbial_concentration_time.title}}'},
+            text: '{{plot_atributes.title}}'},
```
Comment (moonso) [inline vogue/server/templates/microbial.html]:
```suggestion
            text: '{{plot_attributes.title}}'},
```

File: vogue/server/templates/microbial.html [hunk 6]
```diff
@@ -30,11 +30,11 @@
...
         xAxis: {
-            categories: {{ microbial_concentration_time.labels| tojson }}},
+            categories: {{ plot_atributes.labels| tojson }}},
```
Comment (moonso) [inline vogue/server/templates/microbial.html]:
```suggestion
            categories: {{ plot_attributes.labels| tojson }}},
```

File: vogue/server/templates/microbial.html [hunk 7]
```diff
@@ -30,11 +30,11 @@
...
         yAxis: {
-            title: { text: {{ microbial_concentration_time.axis.y | tojson }}}},
+            title: { text: {{ plot_atributes.axis.y | tojson }}}},
```
Comment (moonso) [inline vogue/server/templates/microbial.html]:
```suggestion
            title: { text: {{ plot_attributes.axis.y | tojson }}}},
```

File: vogue/server/templates/wgs.html [hunk 8]
```diff
@@ -81,16 +81,16 @@
...
-    {% if concentration_time.group.no_group is defined %}
-    var data = {{ concentration_time.group.no_group.data | tojson }}
+    {% if concentration_time.all_samples is defined %}
+    var data = {{ concentration_time.all_samples.data | tojson }}
     {% endif %}
 
     Highcharts.chart('concentration_time', {
         chart: { type: 'spline' },
-        title: { text: '{{concentration_time.title}}'},
+        title: { text: '{{plot_atributes.title}}'},
```
Comment (moonso) [inline vogue/server/templates/wgs.html]:
```suggestion
        title: { text: '{{plot_attributes.title}}'},
```

File: vogue/server/templates/wgs.html [hunk 9]
```diff
@@ -81,16 +81,16 @@
...
         subtitle: { text: 'bla bla bla' },
-        xAxis: { categories: {{ concentration_time.labels| tojson }}},
-        yAxis: { title: { text: {{ concentration_time.axis.y | tojson }}}},
+        xAxis: { categories: {{ plot_atributes.labels| tojson }}},
```
Comment (moonso) [inline vogue/server/templates/wgs.html]:
```suggestion
        xAxis: { categories: {{ plot_attributes.labels| tojson }}},
```

File: vogue/server/templates/wgs.html [hunk 10]
```diff
@@ -81,16 +81,16 @@
...
+        yAxis: { title: { text: {{ plot_atributes.axis.y | tojson }}}},
```
Comment (moonso) [inline vogue/server/templates/wgs.html]:
```suggestion
        yAxis: { title: { text: {{ plot_attributes.axis.y | tojson }}}},
```

File: vogue/server/utils.py [hunk 11]
```diff
@@ -3,23 +3,10 @@
 from mongo_adapter import get_client
 from datetime import datetime as dt
 import numpy as np
-from vogue.constants.constants import MONTHS, COLORS
+from vogue.constants.constants import MONTHS, COLORS, TEST_SAMPLES
```
Comment (moonso) [inline vogue/server/utils.py]:
```suggestion
from vogue.constants.constants import (MONTHS, COLORS, TEST_SAMPLES)
```

File: vogue/server/utils.py [hunk 12]
```diff
@@ -58,101 +32,145 @@ def get_percentiles(samples: list, key: str)-> float:
...
+            group['$group'][y_val] = {'$sum': 1}
+        else:
+            # get average of y_val:
+            project['$project'][y_val] = 1
+            group['$group'][y_val] = {'$avg': '$' + y_val}
+ 
+    return [match, project, match_year, group, sort]
+    
+def reformat_aggregate_results(aggregate_result, y_vals, group_key = None):
+    """Reformatses the raw output from the aggregateion query to the format requiered by the plots.
```
Comment (moonso) [inline vogue/server/utils.py]:
```suggestion
    """Reformats raw output from the aggregation query to the format required by the plots.
```

File: vogue/server/utils.py [hunk 13]
```diff
@@ -58,101 +32,145 @@ def get_percentiles(samples: list, key: str)-> float:
...
+
+def value_per_month(adapter, year: str, y_vals: list, group_key: str = None):
+    """Wraper function that will build a pipe, aggregate results from database and 
+    prepare the data for the plots."""
+
+    pipe = pipe_value_per_month(int(year), y_vals, group_key)
+    aggregate_result = adapter.samples_aggregate(pipe)
+    return reformat_aggregate_results(list(aggregate_result), y_vals, group_key)
+
+def plot_atributes( y_axis_label: str, title: str):
```
Comment (moonso) [inline vogue/server/utils.py]:
```suggestion
def plot_attributes( y_axis_label: str, title: str):
```

File: vogue/adapter/plugin.py (hunk 1 above)
Comment (mayabrandi) [reply vogue/adapter/plugin.py]: yep

**Comment by mayabrandi** [comment]
I think I have changed the things you asked for now @moonso

**Pull request merged**